File: src/main/java/com/flaghacker/uttt/common/Player.java
package com.flaghacker.uttt.common;

public enum Player
{
	NEUTRAL(" "),
	PLAYER("X"),
	ENEMY("O"),
	;

	private String niceString;

	Player(String niceString)
	{
		this.niceString = niceString;
	}

	public Player other()
	{
		if (this == PLAYER)
			return ENEMY;
		else if (this == ENEMY)
			return PLAYER;
		else
			throw new IllegalArgumentException("player should be one of [PLAYER, ENEMY]; was " + this);
	}

	public String toNiceString()
	{
		return niceString;
	}

	public static Player fromNiceString(String string)
	{
		for (Player player : values())
			if (player.toNiceString().equals(string))
				return player;

		throw new IllegalArgumentException(string + " is not a valid Player");
	}
}

File: src/main/kotlin/com/flaghacker/sttt/common/Player.kt
package com.flaghacker.sttt.common

enum class Player(val niceString: String) {
	PLAYER("X"),
	ENEMY("O"),
	NEUTRAL(" ");

	fun other(): Player = when(this) {
		PLAYER -> ENEMY
		ENEMY -> PLAYER
		else -> throw IllegalArgumentException("player should be one of [PLAYER, ENEMY]; was " + this)
	}

	fun otherWithNeutral(): Player = if (this == NEUTRAL) NEUTRAL else this.other()

	companion object {
		fun fromNiceString(string: String): Player {
			Player.values().filter { it.niceString == string }.forEach { return it }
			throw IllegalArgumentException("$string is not a valid Player")
		}
	}
}
